packages/block-editor/src/components/dark-editor-style/index.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import tinycolor from 'tinycolor2';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+export default function DarkEditorStyle( ref ) {
+	useEffect( () => {
+		const { ownerDocument } = ref.current;
+		const editorStylesWrapper = ownerDocument.getElementsByClassName(
+			'editor-styles-wrapper'
+		);
+		const backgroundColor = window
+			.getComputedStyle( editorStylesWrapper[ 0 ], null )
+			.getPropertyValue( 'background-color' );
+
+		if ( tinycolor( backgroundColor ).getLuminance() > 0.5 ) {
+			// Is 100 the right number?
+			const body = ownerDocument.getElementsByTagName( 'body' )[ 0 ];
+			// If we only remove it rather than adding it, then if themes don't declare support this will have no impact
+			body.classList.remove( 'is-dark-theme' );
+		}
+	}, [ ref ] );
```

Comment:
We've been moving away from hooks that receive refs as argument to hooks that return refs (using callback refs) (see  useFocusOnMount for an example of this pattern)